### PR TITLE
chore: upgrade shellcheck to 0.9.0

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Upgrading shellcheck, really messy because of how we handle the setup scripts. I'm not 100% sure yet if adding error checking (`set -euo pipefail`) is correct for all of these scripts, I'm keeping this as draft until I verify.

### Checklist

- [X] Changelog updated or skip changelog label added

### CODE REVIEW

File: ci/_build_functions.sh [hunk 1]
```diff
@@ -2,15 +2,18 @@
...
+set -euo pipefail
+
 err_report() {
     echo "Script error on line $1"
     exit 1
 }
 trap 'err_report $LINENO' ERR
 
 function helm() {
+  PWD=${pwd}
```
Comment (sumo-drosiek) [inline ci/_build_functions.sh:14]:
```suggestion
  PWD="$(pwd)"
```

File: tests/helm/terraform/static/all_fields.output.yaml [hunk 2]
```diff
@@ -39,18 +39,24 @@ data:
...
+    declare -a dirs
+    ls -1 /customer-scripts 2>/dev/null | grep _ | grep -oE '^.*?_' | sed 's/_//g' | sort | uniq | read -ar dirs || true
+    for dir in "${dirs[@]}"; do
       target="/scripts/${dir}"
       mkdir "${target}"
       # shellcheck disable=SC2010
       # Get files for given directory and take only filename part (after first _)
-      for file in $(ls -1 "/customer-scripts/${dir}_"* | grep -oE '_.*' | sed 's/_//g'); do
+      files=$(ls -1 "/customer-scripts/${dir}_"* | grep -oE '_.*' | sed 's/_//g')
+      for file in ${files}; do
```
Comment (sumo-drosiek) [inline tests/helm/terraform/static/all_fields.output.yaml:59]: Why do we behave differently for for files than for dirs?
```
declare -a dirs
...
```

File: tests/helm/terraform/static/collector_fields.output.yaml [hunk 3]
```diff
@@ -39,18 +39,24 @@ data:
...
-    for dir in $(ls -1 /customer-scripts | grep _ | grep -oE '^.*?_' | sed 's/_//g' | sort | uniq); do
+    declare -a dirs
+    ls -1 /customer-scripts 2>/dev/null | grep _ | grep -oE '^.*?_' | sed 's/_//g' | sort | uniq | read -ar dirs || true
+    for dir in "${dirs[@]}"; do
       target="/scripts/${dir}"
       mkdir "${target}"
       # shellcheck disable=SC2010
       # Get files for given directory and take only filename part (after first _)
-      for file in $(ls -1 "/customer-scripts/${dir}_"* | grep -oE '_.*' | sed 's/_//g'); do
+      files=$(ls -1 "/customer-scripts/${dir}_"* | grep -oE '_.*' | sed 's/_//g')
```
Comment (sumo-drosiek) [inline tests/helm/terraform/static/collector_fields.output.yaml:58]: again different behavior for files and dirs

File: vagrant/provision.sh [hunk 4]
```diff
@@ -62,10 +62,11 @@ chmod +x /usr/local/bin/yq-"${YQ_VERSION}"
 ln -s /usr/local/bin/yq-"${YQ_VERSION}" /usr/local/bin/yq
 
 # Install docker
+LSB_RELEASE=$(lsb_release -cs)
```
Comment (sumo-drosiek) [inline vagrant/provision.sh:65]:
```suggestion
LSB_RELEASE="$(lsb_release -cs)"
```

File: vagrant/provision.sh [hunk 5]
```diff
@@ -62,10 +62,11 @@ chmod +x /usr/local/bin/yq-"${YQ_VERSION}"
...
 curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
 add-apt-repository \
    "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
-   $(lsb_release -cs) \
+   ${LSB_RELEASE} \
```
Comment (sumo-drosiek) [inline vagrant/provision.sh:69]:
```suggestion
   "${LSB_RELEASE}" \
```

File: vagrant/scripts/get-kubeconfig.sh [hunk 6]
```diff
@@ -30,7 +30,8 @@ vagrant ssh-config > "${SSH_CONFIG_PATH}"
 ssh -o StrictHostKeyChecking=no -F "${SSH_CONFIG_PATH}" default 'kubectl config view --raw' | \
 	sed "s/127.0.0.1/${VAGRANT_IP}/g" > "${CONFIG_PATH}"
 
-mv -v "${KUBECONFIG_PATH}" "${KUBECONFIG_PATH}.bkp.$(date +%s)"
+TODAY=$(date +%s)
```
Comment (sumo-drosiek) [inline vagrant/scripts/get-kubeconfig.sh:33]:
```suggestion
TODAY="$(date +%s)"
```

File: vagrant/scripts/perf-test.sh [hunk 7]
```diff
@@ -71,7 +71,8 @@ function run_test() {
     local DURATION_SECONDS="${2}"
     echo "Starting perf tests"
 
-    if [[ $(is_avalanche_running) -eq 0 ]]; then
+    IS_AVALANCHE_RUNNING=$(is_avalanche_running)
```
Comment (sumo-drosiek) [inline vagrant/scripts/perf-test.sh:74]:
```suggestion
    IS_AVALANCHE_RUNNING="$(is_avalanche_running)"
```

File: tests/helm/terraform/static/all_fields.output.yaml (hunk 2 above)
Comment (swiatekm-sumo) [reply tests/helm/terraform/static/all_fields.output.yaml:59]: Shellcheck complains about one, but not the other, not sure why.

File: vagrant/provision.sh (hunk 5 above)
Comment (swiatekm-sumo) [reply vagrant/provision.sh:69]: Had to revert this part, as it's already quoted and shellcheck complains about it.

File: tests/helm/terraform/static/all_fields.output.yaml (hunk 2 above)
Comment (swiatekm-sumo) [reply tests/helm/terraform/static/all_fields.output.yaml]: I fixed one and forgot about the other. Done now.

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**